org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/properties/SonarLintProjectPropertyPage.java

```diff
@@ -155,13 +155,13 @@ protected void performDefaults() {
 
   @Override
   public boolean performOk() {
-    final SonarLintProject sonarProject = SonarLintProject.getInstance(getProject());
-    sonarProject.setAutoEnabled(enabledBtn.getSelection());
-    sonarProject.save();
+    final SonarLintProjectConfiguration projectConfig = SonarLintProjectConfiguration.read(getProject().getScopeContext());
+    projectConfig.setAutoEnabled(enabledBtn.getSelection());
+    projectConfig.save();
     return super.performOk();
   }
 
-  private IProject getProject() {
-    return (IProject) getElement().getAdapter(IProject.class);
+  private ISonarLintProject getProject() {
+    return (ISonarLintProject) getElement().getAdapter(ISonarLintProject.class);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "ISonarLintProject". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1905)